packages/next/build/babel/preset.ts

```diff
@@ -176,6 +176,8 @@ module.exports = (
           removeImport: true,
         },
       ],
+      require('@babel/plugin-proposal-optional-chaining'),
```

Comment:
This is wrong as it'll add the plugins for all files instead of just for typescript. You'll want to use `overrides`, similarly to `@babel/preset-typescript` https://github.com/babel/babel/blob/master/packages/babel-preset-typescript/src/index.js#L43-L56

Reply:
I think we can use this in JS too, am I wrong?

Reply:
It would be great to use this in JS too, CRA released for both: https://github.com/facebook/create-react-app/releases/tag/v3.3.0. PR: https://github.com/facebook/create-react-app/pull/7438

Reply:
We're not planning on supporting JS syntax features which are not stage 4, as they're prone to change. TypeScript is different as they've already shipped it unflagged in a stable release.

Reply:
That makes sense, thanks for the explanation @Timer 😊

Reply:
That makes sense, but TypeScript allows us to write future JS. For example class properties, I believe it's in stage 3, but TS has support of it. As you said, this syntaxes are shipped unflagged in a stable release. So yeah, I think we could consider to add support of this syntaxes for JS too. What do you think?

Reply:
Let's not add for JS -- it's not future/real syntax yet. I'll likely go stage 4, at which point we can turn on for JS.

Reply:
I just read this https://twitter.com/drosenwasser/status/1202310742436761600?s=19 does this mean we can add it for JS too?